src/irc_bridge.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the IRC bridge class to represent the matrix-appservice-app plugin for Synapse."""
+
+# disabling due the fact that collect status does many checks
+# pylint: disable=too-many-return-statements
+
+import logging
+import typing
+
+import ops
+from ops.pebble import APIError, ExecError
+
+import synapse
+from charm_state import CharmState
+
+logger = logging.getLogger(__name__)
+
+IRC_SERVICE_NAME = "irc"
+
+
+class PEMCreateError(Exception):
+    """An exception raised when the PEM file creation fails."""
+
+    def __init__(self, message: str):
+        """Initialize a new instance of the PEMCreateError class.
+
+        Args:
+            message: The error message.
+        """
+        super().__init__(message)
+
+
+class IRCBridge(ops.Object):  # pylint: disable=too-few-public-methods
+    """A class representing the IRC bridge plugin for Synapse application.
+
+    See https://github.com/matrix-org/matrix-appservice-irc/ for more details about it.
+    """
+
+    def __init__(self, charm: ops.CharmBase, charm_state: CharmState):
+        """Initialize a new instance of the IRC bridge class.
+
+        Args:
+            charm: The charm object that the IRC bridge instance belongs to.
+            charm_state: Instance of CharmState.
+        """
+        super().__init__(charm, "irc-bridge")
+        self._charm = charm
+        self._charm_state = charm_state
+        self.framework.observe(charm.on.collect_unit_status, self._on_collect_status)
+
+    @property
+    def _pebble_service(self) -> typing.Any:
```

Comment:
Is it possible to define a type here that is more specific than `Any`?

Reply:
Yup, good call, turned it into a PebbleService